AirCasting/CoreData/ExternalSessionsStore.swift

```diff
@@ -0,0 +1,113 @@
+// Created by Lunar on 26/04/2022.
+//
+
+import Foundation
+import CoreData
+import CoreLocation
+
+protocol ExternalSessionsStore {
+    func createExternalSession(session: ExternalSessionWithStreamsAndMeasurements, completion: @escaping (Result<Void, Error>) -> Void)
+    func doesSessionExist(uuid: String) -> Bool
+    func deleteSession(uuid: String, completion: @escaping (Result<Void, Error>) -> Void)
+}
+
+struct DefaultExternalSessionsStore: ExternalSessionsStore {
+    private let context: NSManagedObjectContext
+    
+    init(context: NSManagedObjectContext) {
+        self.context = context
+    }
+    
+    func createExternalSession(session: ExternalSessionWithStreamsAndMeasurements, completion: @escaping (Result<Void, Error>) -> Void) {
+        enum CreatingExternalSessionError: Error {
+            case sessionAlreadyExists
+        }
+        
+        context.perform {
+            do {
+                guard (try? context.existingExternalSession(uuid: session.uuid)) == nil else {
+                    throw CreatingExternalSessionError.sessionAlreadyExists
+                }
+                
+                let sessionEntity = newSessionEntity()
+                updateSessionsParams(sessionEntity, session: session)
+                session.streams.forEach { stream in
+                    addStream(stream, to: sessionEntity)
+                }
+                
+                // TODO: If thresholds don't exist in the db already, create the thresholdsEntity
+                
+                try context.save()
+                completion(.success(()))
+            } catch {
+                completion(.failure(error))
+            }
+        }
+    }
+    
+    func doesSessionExist(uuid: String) -> Bool {
+        (try? context.existingExternalSession(uuid: uuid)) != nil
```

Comment:
I think this should also be executed on the context queue. We can do that synchronously if we want using `performAndWait` (and store value to return in a helper variable outside of this block)